src/devices/OneWire/FamilyBindings/OneWireThermometerDevice.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading.Tasks;
+
+namespace Iot.Device.OneWire
+{
+    public partial class OneWireThermometerDevice : OneWireDevice
+    {
+        protected internal OneWireThermometerDevice(OneWireBus bus, string deviceId, int family)
+            : base(bus, deviceId, family)
+        {
+        }
+
+        public Task<float> ReadTemperatureAsync()
```

Comment:
Please make sure to read our conventions https://github.com/dotnet/iot/pull/284 (still WIP but should give rough idea of the expectations). TL;DR is that we currently prefer property `Temperature` which returns `double`. We do not have yet opinion on having or not having async APIs so fine to keep it along the property (the return type should be double though)

Reply:
Will fix double, but async is needed. It can take up to 1 sec to read one sensor. I usually go by MS WinRT standards of anything over 50ms is async.

Reply:
fair enough - usually it is much less than 1ms - btw. any chance we could reduce that timing? If you don't have ideas I can tale a look later but I have some flashbacks from years ago that 1-wire requires some large delays so you might be right. Let's leave it as is and I'll bring this up next time I'll be talking about conventions and update the doc accordingly

Reply:
Changed to Units.Temperature